processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
+	// metric name update
+	if transform.NewName != "" {
+		metric.MetricDescriptor.Name = transform.NewName
+	}
+
+	for _, op := range transform.Operations {
+		switch op.Action {
+		case UpdateLabel:
+			mtp.updateLabelOp(metric, op)
+		case AggregateLabels, AggregateLabelValues:
+			mtp.aggregateOp(metric, op)
+		}
+	}
+}
+
+// updateLabelOp updates labels and label values in metric based on given operation
+func (mtp *metricsTransformProcessor) updateLabelOp(metric *metricspb.Metric, op Operation) {
+	for idx, label := range metric.MetricDescriptor.LabelKeys {
+		if label.Key != op.Label {
+			continue
+		}
+		// label key update
+		if op.NewLabel != "" {
+			label.Key = op.NewLabel
+		}
+		// label value update
+		labelValuesMapping := mtp.createLabelValueMapping(op.ValueActions)
+		for _, timeseries := range metric.Timeseries {
+			newValue, ok := labelValuesMapping[timeseries.LabelValues[idx].Value]
+			if ok {
+				timeseries.LabelValues[idx].Value = newValue
+			}
+		}
+	}
+}
+
+// aggregateOp aggregates the data points in metric based on given operation
+// This is shared by aggregation across labels and aggregation across label values as they have similar aggragation processes
+func (mtp *metricsTransformProcessor) aggregateOp(metric *metricspb.Metric, op Operation) {
+	// labelSet is a set of labels to select
+	var labelSet map[string]bool
```

Comment:
Similarly, this conversion is only needed if the operation is a aggregation, and also since the labeSet is a field for aggregation operation, I see this step step being here as the most straightforward. Let me know if I am missing anything.